src/services/bunker_cases/midterm_slashing_penalty.py

```diff
@@ -249,6 +331,11 @@ def get_frame_by_epoch(epoch: EpochNumber, frame_config: FrameConfig) -> FrameNu
         """Get oracle report frame index by epoch"""
         return FrameNumber((epoch - frame_config.initial_epoch) // frame_config.epochs_per_frame)
 
+    @staticmethod
+    def get_ref_epoch_by_frame(frame_number: FrameNumber, frame_config: FrameConfig) -> EpochNumber:
```

Comment:
Duplicated, check Web3Converter class

Reply:
If it is painful to provide frame and chain configs move some methods from class